src/main/java/org/terasology/workstationInGameHelp/systems/InputProcessesHelpItem.java

```diff
@@ -83,98 +71,47 @@ public String getCategory() {
-     * Gets the description of the workstation process. 
-     *
-     * @param workstationProcess the workstation process to get the description from. 
-     * @return a list of paragraph data for workstationProcess.
-     */
-    static List<ParagraphData> getWorkStationProcessParagraphData(WorkstationProcess workstationProcess) {
-        List<ParagraphData> processParagraphData = Lists.newLinkedList();
-
-        FlowParagraphData paragraphData = new FlowParagraphData(null);
-        TextureRegion plusTexture = Assets.getTextureRegion("workstation:plus").get();
-        ImageFlowRenderable plus = new ImageFlowRenderable(plusTexture, plusTexture.getWidth(), plusTexture.getWidth(), null);
-        TextureRegion eqTexture = Assets.getTextureRegion("workstation:equals").get();
-        ImageFlowRenderable eq = new ImageFlowRenderable(eqTexture, eqTexture.getWidth(), eqTexture.getWidth(), null);
-
-        if (workstationProcess instanceof DescribeProcess) {
-            DescribeProcess describeProcess = (DescribeProcess) workstationProcess;
-            boolean isFirst = true;
-            // add all input widgets
-            for (ProcessPartDescription inputDesc : describeProcess.getInputDescriptions()) {
-                if (!isFirst) {
-                    paragraphData.append(plus);
-                }
-                isFirst = false;
-                String hyperlink = inputDesc.getResourceUrn() != null ? inputDesc.getResourceUrn().toString() : null;
-                paragraphData.append(new WidgetFlowRenderable(inputDesc.getWidget(), 48, 48, hyperlink));
-            }
-
-            // add the equals separator
-            paragraphData.append(eq);
-
-            // add the output widgets
-            isFirst = true;
-            for (ProcessPartDescription outputDesc : describeProcess.getOutputDescriptions()) {
-                if (!isFirst) {
-                    paragraphData.append(plus);
-                }
-                isFirst = false;
-                String hyperlink = outputDesc.getResourceUrn() != null ? outputDesc.getResourceUrn().toString() : null;
-                paragraphData.append(new WidgetFlowRenderable(outputDesc.getWidget(), 32, 32, hyperlink));
-            }
-        } else {
-            paragraphData.append(new TextFlowRenderable(workstationProcess.getId() + " cannot be displayed", null, null));
-        }
-
-        processParagraphData.add(paragraphData);
-        return processParagraphData;
+        Collection<WorkstationProcess> relatedWorkstationProcesses =
+                workstationProcessRelatedAssetCache.getInputRelatedWorkstationProcesses(resourceUrn);
+
+        //TODO: group by `workstationProcess.getProcessType()` and add title for the process?
+        /*
+        FlowParagraphData titleParagraphData = new FlowParagraphData(null);
+        titleParagraphData.append(new TextFlowRenderable(workstationProcess.getProcessTypeName(), null, null));
+        result.add(titleParagraphData);
+        */
```

Comment:
What about this?

Reply:
Discussed offline: Grouping would help to understand with which machine a recipe can be used. Giving the process a title could additionally or alternatively to grouping by process type help to understand what's going on, where and what for to do it.